src/sentry/api/endpoints/artifact_lookup.py

```diff
@@ -267,27 +216,39 @@ def get_artifact_bundles_containing_debug_id(
     )
 
 
-def get_release_artifacts(
+def get_artifact_bundles_containing_url(
+    project: Project, release_name: str, dist_name: str, url: str
+) -> Set[Tuple[int, datetime]]:
+    return set(
+        ArtifactBundleIndex.objects.filter(
+            organization_id=project.organization.id,
+            release_name=release_name,
+            dist_name=dist_name,
+            url=url,
+            artifact_bundle__projectartifactbundle__project_id=project.id,
+        ).values_list("artifact_bundle_id", "date_added")[:1]
```

Comment:
Good stuff! This query should partially use the composite index which is nice. We will have to discuss read on repair though, which could be quite interesting.

Reply:
I pushed a new commit that has detection and optional deletion of duplicates on reads.